drivers/comparator/comparator_silabs_acmp.c

```diff
@@ -0,0 +1,184 @@
+{
+	const struct acmp_config *config = dev->config;
+
+	if (ACMP_IntGetEnabled(config->reg)) {
+		return 1;
+	}
+
+	return 0;
+}
+
+static void acmp_irq_handler(const struct device *dev)
+{
+	const struct acmp_config *config = dev->config;
+	struct acmp_data *data = dev->data;
+
+	ACMP_IntClear(config->reg, (uint32_t)(ACMP_IF_RISE | ACMP_IF_FALL));
+
+	if (data->callback == NULL) {
+		return;
+	}
+
+	data->callback(dev, data->user_data);
+}
+
+#define ACMP_REGISTER_CLOCK_INIT_FN_SYM(inst) _CONCAT(acmp_reg_clock_init, inst)
+#define ACMP_REGISTER_CLOCK_INIT_FN_DEFINE(inst)                                                   \
+	static void ACMP_REGISTER_CLOCK_INIT_FN_SYM(inst)(void)                                    \
+	{                                                                                          \
+		CMU_ClockEnable(_CONCAT(cmuClock_ACMP, inst), true);                               \
+	}
+
+#define ACMP_IRQ_HANDLER_INIT_FN_SYM(inst) _CONCAT(acmp_irq_init, inst)
+#define ACMP_IRQ_HANDLER_INIT_FN_DEFINE(inst)                                                      \
+	static void ACMP_IRQ_HANDLER_INIT_FN_SYM(inst)(void)                                       \
+	{                                                                                          \
+		IRQ_CONNECT(DT_INST_IRQN(inst), DT_INST_IRQ(inst, priority), acmp_irq_handler,     \
+			    DEVICE_DT_INST_GET(inst), 0);                                          \
+                                                                                                   \
+		irq_enable(DT_INST_IRQN(inst));                                                    \
+	}
+
+static DEVICE_API(comparator, acmp_api) = {
+	.get_output = acmp_get_output,
+	.set_trigger = acmp_set_trigger,
+	.set_trigger_callback = acmp_set_trigger_callback,
+	.trigger_is_pending = acmp_trigger_is_pending,
+};
+
+#define ACMP_DEVICE(inst)                                                                          \
+	ACMP_REGISTER_CLOCK_INIT_FN_DEFINE(inst)                                                   \
+	ACMP_IRQ_HANDLER_INIT_FN_DEFINE(inst)                                                      \
+	static struct acmp_data _CONCAT(data, inst);                                               \
+                                                                                                   \
+	static const struct acmp_config _CONCAT(config, inst) = {                                  \
+		.reg = (ACMP_TypeDef *)DT_INST_REG_ADDR(inst),                                     \
+		.reg_clock_init = ACMP_REGISTER_CLOCK_INIT_FN_SYM(inst),                           \
+		.irq_init = ACMP_IRQ_HANDLER_INIT_FN_SYM(inst),                                    \
+		.init = ACMP_DT_INST_CONFIG_INIT(inst),                                            \
+		.negsel =                                                                          \
+			_CONCAT(_ACMP_INPUTCTRL_NEGSEL_, DT_INST_STRING_TOKEN(inst, neg_select)),  \
```

Comment:
If the binding is changed to an integer, it can be directly inserted here using `DT_INST_PROP()`.